compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -288,17 +291,26 @@ object desugar {
     }
 
     val isCaseClass = mods.is(Case) && !mods.is(Module)
+    val isEnum = mods.hasMod[Mod.Enum]
+    val isEnumCase = isLegalEnumCase(cdef)
     val isValueClass = parents.nonEmpty && isAnyVal(parents.head)
       // This is not watertight, but `extends AnyVal` will be replaced by `inline` later.
 
-    val constrTparams = constr1.tparams map toDefParam
+    val originalTparams =
+      if (isEnumCase && parents.isEmpty) {
+        if (constr1.tparams.nonEmpty)
+          ctx.error(em"case with type parameters needs extends clause", constr1.tparams.head.pos)
```

Comment:
What if the parent takes no type parameters, while the child has type parameters, like the following case:
```Scala
enum Base {
  case Bar[T](x: T)
}
```